Create SSL Document Page

eyedarts opened the pull request:
Here's a process for getting SSL working for a local instance of Sandstorm using sandcats.io for DNS and Nginx.

garply commented:
Hey admins! Can one of you let me know if it's safe to build this pull request? See https://alpha.sandstorm.io/grain/kB7bofATTL2jGuKSFwmqdA/GithubPRCommands for a list of commands.

kentonv commented:
@paulproteus will get to this soon, but one thing I noticed: I think the file needs to have the extension `.md` and be formatted as Markdown for the doc site generator to work.

paulproteus commented:
Hi @eyedarts ,

I'd love to merge this into the Sandstorm documentation, since lots of people seem to want to create a setup like this! Thanks for writing it.

I had a few questions & suggestions for you.

* Are you willing to reformat this as a Markdown file, as @kentonv noted? That's the format for the other docs, and I'd love for this to be consistent with that.

* Since this is a substantial contribution to documentation, can I ask you to agree to the Sandstorm CLA to give us permission to share your code? It's here -- https://sandstorm.io/cla/ -- and it's the same form as the Apache Foundation uses, if that helps to know.

* I'd love to see a brief explanation at the top of the document indicating when someone would use this. Are you possibly interested in writing a couple sentences about that, perhaps at the top of the document?

I know it's been a while since you submitted this. Thank you for writing all these docs, as well as for testing them.

If you want me to take it from here and blend this doc into the main docs, I can do that. I'm even happier to help you get this to where it needs to be to land in the official Sandstorm documentation.

paulproteus commented:
To be more precise about to convert this to Markdown, I should say:

* Rename the file from `docs/administering/ssl` to `docs/administering/ssl.md`

* Add a note to `mkdocs.yml` that says where to place the file in the table of contents.

* Render the `docs/administering/ssl.md` file to HTML and make sure it looks good.

There are two options for that last step. 

(1) Easy -- use the GitHub "View" button on the pull request or "Preview" in their web-based editor.

(2) The way I do it -- install a tool called `mkdocs` via http://www.mkdocs.org/ and run `mkdocs serve` from the top level of the Sandstorm repository.

eyedarts commented:
Ok, I updated the ssl page to be ssl.md, made some changes to make it more Markdown friendly, added the intro paragraph, and created another pull request for adding the page to mkdocs.yml.

paulproteus commented on docs/administering/ssl.md:
```diff
@@ -0,0 +1,70 @@
+The following is a process for self-hosted instances of Sandstorm to use SSL with sandcats.io DNS. These steps create a Certificate Authority (CA), corresponding CA Certificate, and the private and public keys for the `[anything].sandcats.io` and `*.[anything].sandcats.io` domains.
```
Can I suggest you add the following note underneath this paragraph:

**Note**: Web browsers will display a big red certificate error when you try to connect to this install. This tutorial is appropriate if you are OK with reconfiguring web browsers to trust a custom certificate authority that you will create during this tutorial. For automatically-trusted SSL configuration, you will need to buy a certificate from a certificate authority.

paulproteus replied:
That way, people who see this tutorial will know that it might not be appropriate for their needs, and those who want this behavior will be happy there is documentation on how to set it up.

eyedarts replied:
Done.

paulproteus commented:
I'm going to merge this! Thanks so much for writing this!

eyedarts commented:
You are very welcome. Thank **YOU** and the others working on Sandstorm for creating this very helpful software!
